common/src/main/java/com/wynntils/handlers/scoreboard/ScoreboardHandler.java

```diff
@@ -56,41 +55,35 @@ public void addPart(ScoreboardPart scoreboardPart) {
         scoreboardParts.add(scoreboardPart);
     }
 
-    private String getScoreboardName() {
-        if (scoreboardNameCache != null) {
-            return scoreboardNameCache;
-        }
-
-        String baseName = "sb" + McUtils.player().getScoreboardName();
+    private boolean isValidScoreboardName(String scoreboardName) {
+        // If the name is longer than 14 characters, we need to trim it (16 chars max, 2 reversed for sb/bf)
+        String name = McUtils.player().getScoreboardName();
+        name = name.length() > 14 ? name.substring(0, 14) : name;
 
-        // If the baseName is longer than 16 characters, we need to trim it
-        scoreboardNameCache = baseName.length() > 16 ? baseName.substring(0, 16) : baseName;
-        return scoreboardNameCache;
+        return (scoreboardName.startsWith("sb") || scoreboardName.startsWith("bf")) && scoreboardName.endsWith(name);
```

Comment:
Is it okay that we only have a SCOREBOARD_KEY for SB? (Still not through understanding the logic, just noting the inconsistency)

Reply:
Yes, scoreboard key is just a third scoreboard the mod creates to display. We don't do any switcheroos, so a static key is fine.

Reply:
(Our "switcheroos" logic is handled by wynn, we just copy what they would display.)